### DISCUSSION TITLE
Emails do not close after being scheduled

### DISCUSSION THREAD

**Discussion by Kzazam** [Extended-Thunder/send-later#369] [General]
Hi everyone, 

Just set up a new computer and have an issue with "Send later" that I never had before. Maybe someone can help, otherwise it's a bug. 

When I schedule an e-mail to be send later (using any of the available options), it e-mail editor does not close automatically. 
The e-mail is saved in the drafts folder. The e-mail is being send at the scheduled time. But the e-mail editor window does not close automatically after I scheduled the e-mail to be send later. The editor window stays open.
I can close the editor window and click "discard". This will close the editor, leave the email in drafts. And the e-mail is sent correctly.
So basically, "Send later" does what it is supposed to do. But the damned editor window does not close automatically. Of course, I can close the editor manually. But it used to close automatically, and that's of course more convenient.

Using Thunderbird 91.0.3 (64-bit)
Send later 8.9.5
Windows 10 Pro 64-bit

Any help out there?
Roland

**Comment by jph-sendlater** [comment]
I'm assuming this started after Thunderbird upgraded to version 91 (within the past day or two)?

Edit: I just found out that Thunderbird isn't shipping 91 as an update yet (only offering it as the download option on their website), so it sounds like you setup this computer within the past week. I haven't had my coffee yet, so give me a minute to respond with a more helpful message :sleeping:

**Comment by jph-sendlater** [comment]
Ok, even being more awake, I'm not sure why this would be happening. Is this happening every time, or is it only intermittent? Does it happen with freshly composed messages, or only replies and forwards? Do you notice any other details that might be related?

The first thing to check would be the debug log. Try opening an error console (Ctrl+Shift+J), and go through the process of scheduling a message. The error console is always pretty verbose, but let me know what you see there.

**Comment by Kzazam** [comment]
Sorry, had to get a bit of work done.

On the old machine, everything is fine with Thunderbird 78.13.0 (32-bit) and "Send later" 8.9.5.
The new machine, I did indeed set up in the last few days, and I downloaded Thunderbird 91.0.3. (64-bit) for that and "Send later" 8.9.5.
Both computers run on Windows 10 Pro 64-bit.
No difference betwen replies, forwards or new e-mails.
​Had a few issues with the new VPN in the last couple of days, so I turned that off, too.
I have no further useful observations on top of what I reported in the first message. 

Error console shows error mesaages. Let me just copy all of the relevant lines, since I do not know what is relevant:

INFO [SendLater]: Opening popup static.js:51:13
INFO [SendLater]: Scheduling send later: 51 with options 
Object { sendAt: Date Tue Aug 31 2021 23:16:00 GMT+0200 (Mitteleuropäische Sommerzeit), recurSpec: "none", args: undefined, cancelOnReply: undefined }
static.js:51:13
Prompter: internal dialogs not available in this context. Falling back to window prompt. Prompter.jsm:1084
Uncaught 
Exception { name: "NS_ERROR_FAILURE", message: "Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [nsIEditor.getInlineSpellChecker]", result: 2147500037, filename: "chrome://messenger/content/messengercompose/MsgComposeCommands.js", lineNumber: 401, columnNumber: 0, data: null, stack: "updateEditableFields@chrome://messenger/content/messengercompose/MsgComposeCommands.js:401:38\nToggleWindowLock@chrome://messenger/content/messengercompose/MsgComposeCommands.js:2654:23\nComposeProcessDone@chrome://messenger/content/messengercompose/MsgComposeCommands.js:680:21\n_doFcc2/<@resource:///modules/MessageSend.jsm:962:12\nnotify@resource://gre/modules/Timer.jsm:62:17\n", location: XPCWrappedNative_NoHelper }
MsgComposeCommands.js:401
Prompter: internal dialogs not available in this context. Falling back to window prompt. Prompter.jsm:1084

Can you make sense of any of that?
Cheers

**Comment by jph-sendlater** [comment]
I don't recognize the line about `getInlineSpellChecker`, but it's not clear whether that's the actual problem. Give the following a try:

* Compose a message to yourself, but don't send it.
* Open the error console.
* At the bottom of the error console, enter the following (just copy/paste all three lines at once), and press enter:

```
cw = Services.wm.getMostRecentWindow("msgcompose");
cw.gCloseWindowAfterSave = true;
cw.GenericSendMessage(Ci.nsIMsgCompDeliverMode.SaveAsDraft);
```

The message you just composed should get saved to drafts, and the window should close itself.

**Comment by Kzazam** [comment]
No, does not close. Here's the text from the console:

cw = Services.wm.getMostRecentWindow("msgcompose");
cw.gCloseWindowAfterSave = true;
cw.CompleteGenericSendMessage(Ci.nsIMsgCompDeliverMode.SaveAsDraft);

Prompter: internal dialogs not available in this context. Falling back to window prompt. Prompter.jsm:1084
Promise { <state>: "pending" }

Uncaught 
Exception { name: "NS_ERROR_FAILURE", message: "Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [nsIEditor.getInlineSpellChecker]", result: 2147500037, filename: "chrome://messenger/content/messengercompose/MsgComposeCommands.js", lineNumber: 401, columnNumber: 0, data: null, stack: "updateEditableFields@chrome://messenger/content/messengercompose/MsgComposeCommands.js:401:38\nToggleWindowLock@chrome://messenger/content/messengercompose/MsgComposeCommands.js:2654:23\nComposeProcessDone@chrome://messenger/content/messengercompose/MsgComposeCommands.js:680:21\n_doFcc2/<@resource:///modules/MessageSend.jsm:962:12\nnotify@resource://gre/modules/Timer.jsm:62:17\n", location: XPCWrappedNative_NoHelper }
MsgComposeCommands.js:401

**Comment by Kzazam** [comment]
One more observation: 
If I use "send now" in the "send later" panel, the mail is sent and closed.

**Comment by jph-sendlater** [comment]
Ah, I tracked it down! It's caused by this bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1726280 which is addressed in Thunderbird beta, but not yet in the version 91 release branch (although I seriously hope they fix it before the ESR release goes live, or I'm going to be getting a lot of users running into this).

The problem is that you don't have a spell check dictionary installed, and Thunderbird throws an error and gives up when it really should just ignore it and move on. You should be able to fix that by setting your UI language (Preferences > Composition > Spelling > Language).

**Comment by Kzazam** [comment]
Yep. That was it.  Downloaded two language packs, picked a language in the settings. Works fine now as far as I have tested. I assume the issue is fixed. If it pops up again, I will post again.

Thanks a lot! I love the add-on so unreasonably much, it would have been a bummer not to be able to use it.

**Comment by jph-sendlater** [comment]
:+1: Great. I'm glad it's working!
